addons/account/tests/test_account_move_out_invoice.py

```diff
@@ -3590,3 +3590,36 @@ def test_invoice_mass_posting(self):
             self.assertEqual(aml.partner_id, self.partner_a)
         for aml in invoice2.line_ids:
             self.assertEqual(aml.partner_id, self.partner_b)
+
+    def test_invoice_journal_account_check_constrains(self):
+        """
+        Test account-journal constraint check is working as expected
+        """
```

Comment:
I think we should add straight forward explanation context like:
- `journal_a` accepts `account_a` but not `account_b`
- `journal_b` accepts `account_b` but not `account_a`
- changing account and journal at the same time should work
- changing only account or only journal should raise an error